docs/user/tutorial-asgi.rst

```diff
@@ -0,0 +1,1090 @@
+            image = await self.store.save(image_id, data)
+
+            resp.location = image.uri
+            resp.media = image.serialize()
+            resp.status = falcon.HTTP_201
+
+
+    class Thumbnails:
+
+        def __init__(self, store):
+            self.store = store
+
+        async def on_get(self, req, resp, image_id, width, height):
+            image = self.store.get(str(image_id))
+            if not image:
+                raise falcon.HTTPNotFound
+            if req.path not in image.thumbnails():
+                raise falcon.HTTPNotFound
+
+            resp.content_type = falcon.MEDIA_JPEG
+            resp.data = await self.store.make_thumbnail(image, (width, height))
+
+Adding a new thumbnails route in ``app.py`` is left as an exercise for the
+reader (if you get stuck, see the final version of ``app.py`` later in this
+tutorial).
+
+The new ``thumbnails`` end-point should now render thumbnails on-the-fly::
+
+  $ http POST localhost:8000/images @/home/user/Pictures/test.png
+
+  HTTP/1.1 201 Created
+  content-length: 319
+  content-type: application/json
+  date: Tue, 24 Dec 2019 18:58:20 GMT
+  location: /images/f2375273-8049-4b10-b17e-8851db9ac7af.jpeg
+  server: uvicorn
+
+  {
+      "id": "f2375273-8049-4b10-b17e-8851db9ac7af",
+      "image": "/images/f2375273-8049-4b10-b17e-8851db9ac7af.jpeg",
+      "modified": "Tue, 24 Dec 2019 18:58:21 GMT",
+      "size": [
+          462,
+          462
+      ],
+      "thumbnails": [
+          "/thumbnails/f2375273-8049-4b10-b17e-8851db9ac7af/231x231.jpeg",
+          "/thumbnails/f2375273-8049-4b10-b17e-8851db9ac7af/115x115.jpeg"
+      ]
+  }
+
+
+  $ http localhost:8000/thumbnails/f2375273-8049-4b10-b17e-8851db9ac7af/115x115.jpeg
+
+  HTTP/1.1 200 OK
+  content-length: 2985
+  content-type: image/jpeg
+  date: Tue, 24 Dec 2019 19:00:14 GMT
+  server: uvicorn
+
+  +-----------------------------------------+
+  | NOTE: binary data not shown in terminal |
+  +-----------------------------------------+
+
+Again, we could also verify thumbnail URIs in the browser or image viewer that
+supports HTTP input.
+
+.. _asgi_tutorial_caching:
+
+Caching Responses
+-----------------
+
+Although scaling thumbnails on-the-fly sounds cool, and we also avoid many pesky
+small files littering our storage, it consumes CPU resources, and we would
+soon find our application crumbling under load.
+
+Let's thus implement response caching in Redis, utilizing
+`aioredis <https://github.com/aio-libs/aioredis>`_ for async support::
+
+  pip install aioredis
+
+We will also need to serialize response data (the ``Content-Type`` header and
+the body in the first version); ``msgpack`` should do::
+
+  pip install msgpack
+
+Our application will obviously need access to a Redis server. Apart from just
+installing Redis server on your machine, one could also:
+
+* Spin up Redis in Docker, eg::
+
+    docker run -p 6379:6379 redis
+
+* Considering Redis is installed on the machine, one could also try
```

Comment:
Considering does not sound right here, but I don't have a better option

Reply:
I can change to _Assuming_.